### PR TITLE
feat: add `openNewPublication` dispatch method

### PR DESCRIPTION
**Pull request by ruslan-sazonov**
This PR adds a new definition of the `appBridge.openNewPublication()` method which will be used in a new TemplateBlock.

### CODE REVIEW

File: packages/app-bridge/src/AppBridgeBlock.ts [hunk 1]
```diff
@@ -89,6 +90,12 @@ export interface AppBridgeBlock extends AppBridgeBase {
      */
     closeTemplateChooser(): void;
 
+    /**
+     * @deprecated This will be removed in version 4.0.0 of @frontify/app-bridge
+     * Use appBridge.dispatch(createNewPublication()) instead
+     */
+    createNewPublication(options: CreateNewPublicationOptions): void;
+
```
Comment (SamuelAlev) [inline packages/app-bridge/src/AppBridgeBlock.ts:98]:
```suggestion
```

File: packages/app-bridge/src/types/Terrific.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ export type TemplateChooserResult = TemplateApiLegacy;
 
 export type AssetChooserAssetChosenCallback = (selectedAsset: { screenData: AssetChooserResult[] }) => void;
 export type TemplateChooserTemplateChosenCallback = (selectedTemplate: { template: TemplateChooserResult }) => void;
+export type CreateNewPublicationOptions = {
+    template: Template;
+    projectId: number;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/Terrific.ts:27]: Please have a look to the web-app comment

File: packages/app-bridge/src/AppBridgeBlock.ts (hunk 1 above)
Comment (ruslan-sazonov) [reply packages/app-bridge/src/AppBridgeBlock.ts]: Cleaned up

File: packages/app-bridge/src/AppBridgeBlock.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import type {
     ColorPaletteCreate,
     ColorPalettePatch,
     ColorPatch,
+    CreateNewPublicationOptions,
```
[the remark this answers is not in the document]
Comment (ruslan-sazonov) [reply packages/app-bridge/src/AppBridgeBlock.ts]: Same here

File: packages/app-bridge/src/types/Terrific.ts (hunk 2 above)
Comment (ruslan-sazonov) [reply packages/app-bridge/src/types/Terrific.ts]: Cleaned up the `projectId` prop. Thanks for hint!

**Review by utmishra — APPROVED** [review]
LGTM 👍🏽

File: packages/app-bridge/src/types/DispatchHandler.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ export type DispatchPayload = {
     openTemplateChooser: void;
     closeTemplateChooser: void;
     openNavigationManager: void;
+    createNewPublication: CreateNewPublicationOptions;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/DispatchHandler.ts:12]:
```suggestion
    openNewPublication: OpenNewPublicationPayload;
```

File: packages/app-bridge/src/registries/commands/index.ts [hunk 5]
```diff
@@ -5,3 +5,4 @@ export * from './AssetViewer';
 export * from './CommandRegistry';
 export * from './NavigationManager';
 export * from './TemplateChooser';
+export * from './CreateNewPublication';
```
Comment (SamuelAlev) [inline packages/app-bridge/src/registries/commands/index.ts:8]: Missing file

Comment (ruslan-sazonov) [reply packages/app-bridge/src/registries/commands/index.ts]: Yep, added a command that aligns the new code structure.

File: packages/app-bridge/src/registries/commands/CommandRegistry.ts [hunk 6]
```diff
@@ -17,4 +18,5 @@ export type CommandRegistry = CommandNameValidator<{
     openTemplateChooser: OpenTemplateChooser;
     closeTemplateChooser: CloseTemplateChooser;
     openNavigationManager: OpenNavigationManager;
+    openNewPublication?: OpenNewPublicationPayload;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/registries/commands/CommandRegistry.ts:21]: Can you really open the modal without parameters?

File: packages/app-bridge/src/registries/commands/NewPublication.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import type { DispatchHandlerParameter } from '../../AppBridge';
+import type { CommandRegistry } from './CommandRegistry';
+
+export const openNewPublication = (
+    options?: CommandRegistry['openNewPublication'],
```
Comment (SamuelAlev) [inline packages/app-bridge/src/registries/commands/NewPublication.ts:7]:
```suggestion
    payload: CommandRegistry['openNewPublication'],
```
Same here can it be `undefined`, also renamed `options` to `payload` to align with line 10

File: packages/app-bridge/src/types/DispatchHandler.ts (hunk 4 above)
Comment (SamuelAlev) [reply packages/app-bridge/src/types/DispatchHandler.ts]: All good thanks for you patience ;)

File: packages/app-bridge/src/registries/commands/CommandRegistry.ts [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 import type { CommandNameValidator } from '../../AppBridge';
 import type { AssetChooserOptions } from '../../types';
+import type { OpenNewPublicationPayload } from '../../types';
```
[the remark this answers is not in the document]
Comment (ruslan-sazonov) [reply packages/app-bridge/src/registries/commands/CommandRegistry.ts]: Make sense. Fixed.

File: packages/app-bridge/src/registries/commands/CommandRegistry.ts (hunk 6 above)
Comment (ruslan-sazonov) [reply packages/app-bridge/src/registries/commands/CommandRegistry.ts]: Actually, no. Fixed.

File: packages/app-bridge/src/registries/commands/NewPublication.ts (hunk 7 above)
Comment (ruslan-sazonov) [reply packages/app-bridge/src/registries/commands/NewPublication.ts]: Fixed 👌

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**